clients/src/main/java/org/apache/kafka/clients/producer/internals/TransactionManager.java

```diff
@@ -1251,6 +1251,9 @@ public void handleResponse(AbstractResponse response) {
                 pendingRequests.add(txnOffsetCommitHandler(result, offsets, builder.consumerGroupId()));
                 transactionStarted = true;
             } else if (error == Errors.COORDINATOR_NOT_AVAILABLE || error == Errors.NOT_COORDINATOR) {
+                lookupCoordinator(FindCoordinatorRequest.CoordinatorType.GROUP, builder.consumerGroupId());
```

Comment:
This is a bit strange. Can you explain why it was needed?

Reply:
handleAddOffsetsToTxnRequest precomputes the offsets topic partition and then handles adding partitions to the transaction. If CoordinatorNotAvailableException is received, `AddOffsetsToTxnHandler` always think it's the transaction coordinator failed. However, `partitionFor` in this patch could also throw `CoordinatorNotAvailableException`, complaining the group coordinator unavailability. That's why I also issue a lookup for group coordinator here.